### ISSUE TITLE
Can't call script function in ASPX FROM .TS in VS2013

### ISSUE THREAD

**Issue by GearTheWorld** [Microsoft/TypeScript#1869]
Hi,

I have a lot of code in my .TS that work just fine.  When I hit F5 in VS2013 everything is very good.  Now I have a function declared in a script tag inside the ASPX and I have to call this function from the .TS file but it doesn't find it and it doesn't compile when I hit F5 I get the error Can't find name Sendquote which is the function declared in the script tag inside the ASPX.

I've been told to write this as a bug but I'm not sure it is a bug si let me know what I'm missing or if it's really a BUG

I'm actually using TypeScript 1.4 in VS 2013 Community edition on a Windows 8.1 Update PC

Thank you

**Comment by nycdotnet** [comment]
This isn't a bug.  If you want TypeScript to know about a function that is declared in a JavaScript file (an embedded JavaScript tag on an ASPX or HTML page counts), you have to define a definition for it.  For example:

```typescript
// I am assuming it takes a string and returns void...
declare function Sendquote(quote: string): void;
```
Your other option is to copy the embedded script to a .ts file and reference it on the ASPX page instead (which is probably the best option in your case).

**Comment by GearTheWorld** [comment]
I did the declaration at the top of the .TS and here's what I got :

http://1drv.ms/1HsoRog

**Comment by GearTheWorld** [comment]
Here is the script in the aspx :

    <script lang="javascript" type="text/javascript">
        function SendQuote(FullName, Email, Phone, Location, Description)
        {
            var Service = new BusinessOperation();
            Service.SendQuote(FullName, Email, Phone, Location, Description, OnSuccess, OnFailure, null);
        }

        function OnSuccess(Result)
        {
            // Do nothing.
        }
        function OnFailure(Error)
        {
            alert(Error.toString());
        }
    </script>

**Comment by GearTheWorld** [comment]
This is the declaration in the .TS at the top :

declare function Sendquote(FullName:string, Email:string, Phone:string, Location:string, Description:string): void;

**Comment by GearTheWorld** [comment]
Arrgggg sorry I saw my mistake.  was the q of the quote wasn't uppercase since in the script it was.

**Comment by GearTheWorld** [comment]
Thank you very much for the hint on how to do this thing.  It's greatly appreciated.

**Comment by GearTheWorld** [comment]
Just a note about the other option you mentioned.  I'm doing this way because I never been able to call PageMethods.MyMethod which is a WebMethod in C# from the .TS directly because all this would have saved me a lot of trouble.  Actually  I had to do a WCF Service which a function can use in a script inside the aspx markup in order to be able to call this function from the .TS

But if you tell me that I can simply call PageMethods.MyMethod which is a C# WebMethod well, this would be awesome since No WCF needed nor a script in the aspx markup

You know I tried this but in .TS PageMethods doesn't exist and I don't know if it's faisable and how.
Of course I have everything setup for that method to be available, scriptmanager having the EnablePageMethods="true" and so on.
